### DISCUSSION TITLE
Web monitoring not working on upstash or I'm just doing something wrong?

### DISCUSSION THREAD

**Discussion by kukicola** [sidekiq/sidekiq#5731] [Q&A]
Hello
I have an app deployed to fly.io and as a result using upstash redis (according to sidekiq dashboard it's redis 6.2.6). I'm using newest sidekiq version and it raises error whenever I enter "retries" or "morgue" pages in web UI. I started to investigate it:

```
get "/retries" do
      @count = (params["count"] || 25).to_i
      (@current_page, @total_size, @retries) = page("retry", params["page"], @count)
      @retries = @retries.map { |msg, score| Sidekiq::SortedEntry.new(nil, score, msg) }

      erb(:retries)
    end
```

On my local machine `@retries` looks like this: `[[job_data, score], [job_data, score]]` but when using upstash its flat: `[job_data, score, job_data, score]`. What cloud be the reason? Upstash missing some features? I compared data returned by `zrange` command via redis-cli and it looks pretty much the same, apart from the fact that in one case score was a float and in the other one a string with float inside

**Comment by kukicola** [comment]
Oh, and I forgot to add information about raised error: `<TypeError: can't convert nil into Float>` https://github.com/mperham/sidekiq/blob/main/lib/sidekiq/web/application.rb#L200

**Comment by mperham** [comment]
Please contact Upstash support.
